source/common/http/conn_manager_impl.cc

```diff
@@ -507,7 +507,7 @@ ConnectionManagerImpl::ActiveStream::ActiveStream(ConnectionManagerImpl& connect
          "Either routeConfigProvider or scopedRouteConfigProvider should be set in "
          "ConnectionManagerImpl.");
   for (const AccessLog::InstanceSharedPtr& access_log : connection_manager_.config_.accessLogs()) {
-    filter_manager_.addAccessLogHandler(access_log);
+    access_log_handlers_.push_back(access_log);
```

Comment:
I'd be inclined to land everything apart from the ownership change apart from the ownership change, but let's make sure we're happy with overall ownership plan first.